docs/common/create_new_rn_feature.md

```diff
@@ -0,0 +1,142 @@
+# Create new RN feature from the scratch
+
+Within the `app` folder create new folder dedicated for your package eg. `custom` and then follow the structure below:
+
+```
+.
+├── app/
+│   │── custom/                    - custom application
+│           │── appRegistry/       - place for RootComponent and CustomStandalonePackage
+│           │── navigation/        - place for StackNavigator
+│           │── scenes/            - folder with custom package screens
+│           │── index.js           - file which is exporting CustomStandalonePackage
+│           └── package.json
+```
+
+## appRegistry
+
+### RootComponent
+
+That file is representing the context providers which are passing data through the component tree for your application without having to pass props down manually at every level.
+
+```js
+<Dimensions.Provider dimensions={this.props.dimensions}>
+  <AuthContext.Provider
+    accessToken={this.props.accessToken || null}
+    bookingId={this.props.bookingId || null}
+    simpleToken={this.props.simpleToken || null}
+  >
+    {this.props.children}
+  </AuthContext.Provider>
+</Dimensions.Provider>
+```
+
+_For more details check [Context documentation]('https://reactjs.org/docs/context.html')_
+
+### CustomStandalonePackage
+
+1. Render stack navigator wrapped in `RootComponent`
+
+2. To avoid problems on native navigation <-> RN navigation export standalone package wrapped within `WithNativeNavigation` HOC
+
+```js
+export default WithNativeNavigation(CustomStandalonePackage, 'Custom');
+```
+
+## navigation
+
+### CustomStack
+
+Within `navigation` folder create your `StackNavigator` which provides a way for your app to transition between screens where each new screen is placed on top of a stack.
+
+The example stack navigator should look like:
+
+```js
+export default StackNavigator(
+  {
+    Custom: {
+      screen: CustomScreen,
+      navigationOptions: {
+        headerLeft: null,
+        ...createTransparentHeaderStyle(RNDimensions.get('screen'))
```

Comment:
What is RNDimensions? Maybe add the import code for it? Is it the `Dimensions` from `react-native` ?

Reply:
I just copy paste it from other file. It's not even needed there - going to remove it.